src/read_write/pts.rs

```diff
@@ -49,21 +58,27 @@ impl Iterator for PtsIterator {
             if parts.len() != 7 {
                 continue;
             }
-            return Some(Point {
-                position: Vector3::new(
-                    parts[0].parse::<f64>().unwrap(),
-                    parts[1].parse::<f64>().unwrap(),
-                    parts[2].parse::<f64>().unwrap(),
-                ),
-                color: Color {
-                    red: parts[4].parse::<u8>().unwrap(),
-                    green: parts[5].parse::<u8>().unwrap(),
-                    blue: parts[6].parse::<u8>().unwrap(),
-                    alpha: 255,
-                },
-                intensity: None,
-            });
+            position.push(Vector3::new(
+                parts[0].parse::<f64>().unwrap(),
+                parts[1].parse::<f64>().unwrap(),
+                parts[2].parse::<f64>().unwrap(),
+            ));
+            color.push(Vector3::new(
+                parts[4].parse::<u8>().unwrap(),
+                parts[5].parse::<u8>().unwrap(),
+                parts[6].parse::<u8>().unwrap(),
+            ));
+            point_count += 1;
+        }
+        if point_count == 0 {
+            None
+        } else {
+            Some(PointsBatch {
+                position,
+                attributes: vec![("color".to_string(), AttributeData::U8Vec3(color))]
```

Comment:
Nitpick: For just one value, it might be preferable to just insert once into a HashMap.

Reply:
I personally like it to have as little mutables as possible. If fine with you, I would like to keep it as it is...